qa/L0_java_memory_growth/Simple.java

```diff
@@ -0,0 +1,1018 @@
+        System.out.println(buffer.limit(byte_size.get()).getString());
+
+        FAIL_IF_ERR(
+            TRITONSERVER_MessageDelete(server_metadata_message),
+            "deleting status metadata");
+      }
+
+      String model_name = "simple";
+
+      // Wait for the model to become available.
+      boolean[] is_torch_model = {false};
+      boolean[] is_int = {true};
+      boolean[] is_ready = {false};
+      health_iters = 0;
+      while (!is_ready[0]) {
+        FAIL_IF_ERR(
+            TRITONSERVER_ServerModelIsReady(
+                server, model_name, 1, is_ready),
+            "unable to get model readiness");
+        if (!is_ready[0]) {
+          if (++health_iters >= 10) {
+            FAIL("model failed to be ready in 10 iterations");
+          }
+          Thread.sleep(500);
+          continue;
+        }
+
+        TRITONSERVER_Message model_metadata_message = new TRITONSERVER_Message(null);
+        FAIL_IF_ERR(
+            TRITONSERVER_ServerModelMetadata(
+                server, model_name, 1, model_metadata_message),
+            "unable to get model metadata message");
+        BytePointer buffer = new BytePointer((Pointer)null);
+        SizeTPointer byte_size = new SizeTPointer(1);
+        FAIL_IF_ERR(
+            TRITONSERVER_MessageSerializeToJson(
+                model_metadata_message, buffer, byte_size),
+            "unable to serialize model status protobuf");
+
+        JsonParser parser = new JsonParser();
+        JsonObject model_metadata = null;
+        try {
+          model_metadata = parser.parse(buffer.limit(byte_size.get()).getString()).getAsJsonObject();
+        } catch (Exception e) {
+          FAIL("error: failed to parse model metadata from JSON: " + e);
+        }
+
+        FAIL_IF_ERR(
+            TRITONSERVER_MessageDelete(model_metadata_message),
+            "deleting status protobuf");
+
+        if (!model_metadata.get("name").getAsString().equals(model_name)) {
+          FAIL("unable to find metadata for model");
+        }
+
+        boolean found_version = false;
+        if (model_metadata.has("versions")) {
+          for (JsonElement version : model_metadata.get("versions").getAsJsonArray()) {
+            if (version.getAsString().equals("1")) {
+              found_version = true;
+              break;
+            }
+          }
+        }
+        if (!found_version) {
+          FAIL("unable to find version 1 status for model");
+        }
+
+        FAIL_IF_ERR(
+            ParseModelMetadata(model_metadata, is_int, is_torch_model),
+            "parsing model metadata");
+      }
+
+      Runnable runnable =
+        () -> { 
+          if(ValidateMemoryGrowth(.1f)){
+            System.out.println("Memory growth test passed");
+          } else {
+            System.out.println("Memory growth test FAILED");
+          }
+        };
+      Thread memory_thread = new Thread(runnable);
+      memory_thread.start();
+
+      for(int i = 0; i < 1000000; i++){
```

Comment:
Can you expose command line option to set the number of iteration? Then we can indirectly control the length of the test in `test.sh`

Reply:
Good recommendation. Done.